store/errors.go

```diff
@@ -137,6 +137,42 @@ type SnapActionError struct {
 	Other []error
 }
 
+// SingleOpError returns the single operation, snap name, and error if
+// e represents a single error of a single operation on a single snap
+// (i.e. if e.Other is empty, and e.Refresh, e.Install and e.Download
+// have a single error in total).
+// In any other case, the error returned will be nil.
+func (e SnapActionError) SingleOpError() (op, name string, err error) {
```

Comment:
this probably needs at least one explicit test even if it's tested indirectly because of the substitution below